GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs

```diff
@@ -785,5 +791,53 @@ private void tsmiCategoryClear_Click(object sender, EventArgs e)
 
             ShowRecentRepositories();
         }
+
+        private void OnDragDrop(object sender, DragEventArgs e)
+        {
+            if (e.Data.GetData(DataFormats.FileDrop) is string[] fileNameArray)
+            {
+                if (fileNameArray.Length != 1)
+                {
+                    return;
+                }
+
+                string dir = fileNameArray[0];
+                if (!string.IsNullOrEmpty(dir) && Directory.Exists(dir))
+                {
+                    GitModule module = new GitModule(dir);
+
+                    if (!module.IsValidGitWorkingDir())
+                    {
+                        DialogResult dialogResult = MessageBox.Show(this, _directoryIsNotAValidRepositoryOpenIt.Text,
+                            _directoryIsNotAValidRepositoryCaption.Text, MessageBoxButtons.YesNo,
+                            MessageBoxIcon.Exclamation, MessageBoxDefaultButton.Button2);
+                        if (dialogResult == DialogResult.No)
```

Comment:
Even if I click "yes", the folder doesn't get opened

Reply:
I think we should just show a message saying "Cannot open the folder" (heading) "Not a valid git repository" (message)